crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -1079,7 +1118,7 @@ impl ScheduleGraph {
     /// - checks for system access conflicts and reports ambiguities
     pub fn build_schedule(
         &mut self,
-        components: &Components,
+        world: &mut World,
```

Comment:
why does this now need &mut World access?

Reply:
Because custom schedule build passes will frequently need `&mut World` access to be useful. This allows us to pass the `&mut World` all the way down. The custom schedule build pass may want to insert resources into the world for these systems.